### ISSUE TITLE
Warnings are presented as errors

### ISSUE THREAD

**Issue by jboesl** [mojohaus/nbm-maven-plugin#6]
This issues is bugging me for a long time already but I can't extract a minimal project where this occurs. I just now ran into it again so I describe what happens.
So I get an error during install:

> [ERROR] Failed to execute goal org.codehaus.mojo:nbm-maven-plugin:3.14:manifest (default-manifest) on project AliasEditor: Module has friend dependency on org.netbeans.core but is not listed as a friend. -> [Help 1] 
> [ERROR]  
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch. 
> [ERROR] Re-run Maven using the -X switch to enable full debug logging. 
> [ERROR]  
> [ERROR] For more information about the errors and possible solutions, please read the following articles: 
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

The issue is that this not the real error. I have to set `<verifyRuntime>warn</verifyRuntime>` in order to see it:

> [WARNING] Module has friend dependency on org.netbeans.core but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.core.startup but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.modules.options.keymap but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.modules.projectui but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.bootstrap but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.modules.favorites but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.modules.masterfs but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.modules.project.indexingbridge but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.modules.versioning.core but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.modules.git but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.modules.editor.settings.storage but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.modules.versioning.util but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.modules.options.editor but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.modules.db.core but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.modules.db.dataview but is not listed as a friend. 
> [WARNING] Module has friend dependency on org.netbeans.modules.db.metadata.model but is not listed as a friend. 
> [INFO] Private classes referenced in module: [de.adito.aditoweb.nbm.entityeditor.dataobjects.BaseEntityGroupDataObject] 
> [ERROR] Project depends on packages not accessible at runtime in module de.adito.aditoweb.nbm:EntityEditor:jar:4.4.0

After adding `de.adito.aditoweb.nbm.entityeditor.dataobjects` as a public package everything works fine.

**Comment by mkleint** [comment]
well, does your module declare a dependency on org.netbeans.core explicitly in the pom.xml?

`[WARNING] Module has friend dependency on org.netbeans.core but is not listed as a friend.`

that's an error because typically an runtime your application would fail to start (or at least enable your module(s)) if you are not a friend of that module. So we make sure your build fails fast rather than later (at runtime). I don't see a relationship to the public package declaration though.

`[INFO] Private classes referenced in module: [de.adito.aditoweb.nbm.entityeditor.dataobjects.BaseEntityGroupDataObject] 
[ERROR] Project depends on packages not accessible at runtime in module de.adito.aditoweb.nbm:EntityEditor:jar:4.4.0`

this thing is actually something that can indeed be fixed by declaring public packages. Again at build time we attempt to guard against problems at runtime. Maven dependencies alone cannot entirely express all the possible dependencies among the modules in the netbeans platform runtime environment..

**Comment by jboesl** [comment]
No, it's doesn't declare a dependency on it.
Well I know about friend dependencies. In some modules I'm declaring an implementation dependency because of those. But that module doesn't have any special dependencies.

The nbm-maven-plugin finished successfully until I used a class which wasn't in a public package. After that I got the failure described above. After defining that package as public the failures no longer occurred.

**Comment by mkleint** [comment]
Would you be able to extract the behaviour into a sample project? I've tried creating 2 nbm module projects, omit public packages in prj1 and have prj2 depend on it but didn't get any warnings as you describe. Only the error that a class is referenced but not public. I've tried to throw some more module dependencies onto prj2 but still the same. Thx.

**Comment by jboesl** [comment]
As already stated I'm unable to provide a small project where the behaviour can be observed. It happens in my main project with about 50 modules. Every attempt to reproduce that failed so far.
I know it's really hard to follow a behaviour someone expreriences just by getting told it happens.

**Comment by mkleint** [comment]
ok, I think I've reproduced the problem.
1. declare a dependency on `org-netbeans-modules-git`. Make it an implementation dependency because this thing doesn't declare a public api.
2. create an action extending `org.netbeans.modules.git.ui.actions.SingleRepositoryAction`
3. when you try to build the project, you get 

```
Failed to execute goal org.codehaus.mojo:nbm-maven-plugin:3.14:manifest (default-manifest) on project mavenproject2: Module has friend dependency on org.netbeans.modules.options.editor but is not listed as a friend. -> [Help 1]
```
1. you have to add `<verifyRuntime>warn</verifyRuntime>` to nbm plugin config to actually find out that you are missing a direct dependency on `org-netbeans-modules-versioning`

**Comment by mkleint** [comment]
Should be fixed now.

**Comment by jboesl** [comment]
Wow, that are great news. Thanks a lot.

**Comment by mkleint** [comment]
3.14.1 or 4.0 (the latter now only working with maven 3.0.5+)

**Comment by mcheung63** [comment]
how u fix it?
